gradle/gradle_properties.go

```diff
@@ -0,0 +1,42 @@
+package gradle
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/buildpacks/libcnb"
+)
+
+type PropertiesFile struct {
+	Binding    libcnb.Binding
+	GradleHome string
+}
+
+func (p PropertiesFile) Contribute(layer libcnb.Layer) (libcnb.Layer, error) {
```

Comment:
I see why it's nescessary to use a layer contributor here, so it runs at the right time. It's a bit weird given that it isn't creating a layer, but I see that as additional data points for [feedback to libcnb](https://github.com/buildpacks/libcnb/issues/49) rather than feedback for this PR.